src/plugins/usage_collection/public/components/track_application_view/track_application_view_wrapper.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ReplaySubject } from 'rxjs';
+import React, { FC } from 'react';
+import { TrackApplicationViewComponent } from './track_application_view';
+import { IApplicationUsageTracker } from '../../plugin';
+import { TrackApplicationViewProps } from './types';
+
+/**
+ * React component to track the number of clicks and minutes on screen of the children components.
+ * @param props {@Link TrackApplicationViewProps}
+ * @constructor
+ */
+export const TrackApplicationView: FC<TrackApplicationViewProps> = (props) => {
+  const propsWithTracker = { ...props, applicationUsageTracker$ };
+  return <TrackApplicationViewComponent {...propsWithTracker} />;
+};
+
+/**
+ * Maintain in memory the applicationUsageTracker in the browser.
+ * There should only exist one reporter at a time, so this is not a big deal to have a global variable.
+ * @private
+ */
+const applicationUsageTracker$ = new ReplaySubject<IApplicationUsageTracker>(1);
+
+/**
+ * Lazy setter of the singleton applicationUsageTracker. To be called by the plugin once the reporter is initialised.
+ * @param newApplicationUsageTracker The new applicationUsageTracker to be set.
+ * @private
+ */
+export function setApplicationUsageTracker(newApplicationUsageTracker: IApplicationUsageTracker) {
+  applicationUsageTracker$.next(newApplicationUsageTracker);
+}
```

Comment:
I would personally avoid that kind of module-based setter/getter as much as possible. The consensus when providing 'connected' components is to do so using the provider's start contracts. ( see `navigation` or `savedObjectTagging` for examples)

Also the dependency between the consumers and this component is misleading. With this side effect, you only require dependent plugins to use a `requiredBundle` in the plugin configuration, where the bundle itself is effectively not sufficient: If the `usage_collection` plugin does not initialize, `setApplicationUsageTracker` is never called and the `TrackApplicationView` component will not work.

Reply:
> I would personally avoid that kind of module-based setter/getter as much as possible. The consensus when providing 'connected' components is to do so using the provider's start contracts. ( see `navigation` or `savedObjectTagging` for examples)

IMHO, for the best Dev Experience, if `usageCollection` is not initialised, the "production" code should work. It simply won't collect the usage. The current and proposed solutions allow it the `requiredBundle` allows using the component whether or not the tracker exists. If we follow similar approaches to `navigation` or `savedObjectTagging`, the plugin has to be required (and that's against the recommendation in the [README](https://github.com/elastic/kibana/blob/master/src/plugins/usage_collection/README.md#L19-L27)).

> Also the dependency between the consumers and this component is misleading. With this side effect, you only require dependent plugins to use a `requiredBundle` in the plugin configuration, where the bundle itself is effectively not sufficient: If the `usage_collection` plugin does not initialize, `setApplicationUsageTracker` is never called and the `TrackApplicationView` component will not work.

With the current implementation (the one this PR tries to change), `usageCollection` is declared as `optionalPlugin`, leading to the same behaviour. I tested what happens when the plugin is disabled and it works as expected: showing the "production" code but simply not tracking the stats.

Reply:
I get your point, and I guess that in that specific case (importing a component that may or may not have behavior), it may be the best DX.

I would still want @spalger to confirm that we are alright importing/using **stateful** code via `requiredBundle` (now or in the future)

Reply:
I agree! In case of doubt, ask @spalger!

Reply:
Hey, back from vacation. I agree with @pgayvallet that we should avoid this. Storing application state in the module scope like this is a pattern with a lot of drawbacks and while this specific scenario probably won't see any issues I still don't think the DX of using context or something is bad enough to justify using module scope like this.

If the plugin exposed a `TrackingProvider` component that consumers could wrap their application in, then the `<TrackApplicationView>` component could just pull the `applicationUsageTracker` component from context and the DX would still be better, but the plugin relationship would still be modeled correctly and application state would be stored in the application where it belongs.

I do agree that "just working without having to do anything else" is probably the best DX, but I think that also leads to unrealistic expectations for the DX in other places and I expect if we make this work here we will likely see this pattern spread into places where storing some application state in the module scope is harmful, leads to hard to understand bugs, we've been there before and I'd like to avoid going back there.

Just my two cents.